pkg/cmd/kafka/create/create.go

```diff
@@ -700,35 +723,107 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
 		payload.SetPlan(mapAmsTypeToBackendType(userQuota) + "." + answers.Size)
 
 	}
-	if opts.useEnterpriseCluster {
-		if len(orgQuota.EnterpriseQuotas) == 0 {
-			return nil, fmt.Errorf("No enterprise quota available")
+	return payload, nil
+}
+
+func marketplaceQuotaPrompt(orgQuota *accountmgmtutil.OrgQuotas, answers *promptAnswers, f *factory.Factory, err error) (*promptAnswers, error) {
+	validMarketPlaces := FetchValidMarketplaces(orgQuota.MarketplaceQuotas, answers.CloudProvider)
+	if len(validMarketPlaces) == 1 {
+		answers.Marketplace = validMarketPlaces[0]
+	} else {
+		marketplacePrompt := &survey.Select{
+			Message: f.Localizer.MustLocalize("kafka.create.input.marketplace.message"),
+			Options: validMarketPlaces,
 		}
-		// casting to nullable string
-		var clusterIdSNS kafkamgmtclient.NullableString
-		clusterIdSNS.Set(opts.selectedCluster.ClusterId)
+		err = survey.AskOne(marketplacePrompt, &answers.Marketplace)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-		var billingModelEnterprise kafkamgmtclient.NullableString
-		var enterprise = "enterprise"
-		billingModelEnterprise.Set(&enterprise)
+	if len(validMarketPlaces) > 0 {
 
-		var nullString kafkamgmtclient.NullableString
-		payload = &kafkamgmtclient.KafkaRequestPayload{
-			Name:                  answers.Name,
-			BillingModel:          billingModelEnterprise,
-			BillingCloudAccountId: nullString,
-			Marketplace:           nullString,
-			ClusterId:             clusterIdSNS,
-			//	set plan here
+		validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+
+		if len(validMarketplaceAcctIDs) == 1 {
+			answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
+		} else {
+			marketplaceAccountPrompt := &survey.Select{
+				Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
+				Options: validMarketplaceAcctIDs,
+			}
+			err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
+			if err != nil {
+				return nil, err
+			}
 		}
 	}
-	return payload, nil
+	return answers, nil
+}
+
+func cloudProviderPrompt(f *factory.Factory, answers *promptAnswers) (*promptAnswers, error) {
+	cloudProviderNames, err := GetEnabledCloudProviderNames(f)
+	if err != nil {
+		return nil, err
+	}
+
+	cloudProviderPrompt := &survey.Select{
+		Message: f.Localizer.MustLocalize("kafka.create.input.cloudProvider.message"),
+		Options: cloudProviderNames,
+	}
+
+	err = survey.AskOne(cloudProviderPrompt, &answers.CloudProvider)
+	if err != nil {
+		return nil, err
+	}
+	return answers, nil
+}
+
+// This may need to altered as the `name` are mutable on ocm side
```

Comment:
is this comment updated? I see that we are using `ID`

Reply:
we are using both, the cluster name and the cluster id in this format:
`clustername (clusterId)`
The issue is that ocm name is a mutable field, however, we're doing the same thing that console.dot and the ocm cli do